airflow/contrib/operators/aws_lambda_operator.py

```diff
@@ -0,0 +1,131 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.contrib.hooks.aws_lambda_hook import AwsLambdaHook
+
+
+class AwsLambdaExecutionError(Exception):
+    """
+    Raised when there is an error executing the function.
+    """
+
+    pass
+
+
+class AwsLambdaPayloadError(Exception):
+    """
+    Raised when there is an error with the Payload object in the response.
+    """
+
+    pass
+
+
+class AwsLambdaInvokeFunctionOperator(BaseOperator):
+    """
+    Invoke AWS Lambda functions with a JSON payload.
+
+    The check_success_function signature should be a single param which will receive a dict.
+    The dict will be the "Response Structure" described in
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html#Lambda.Client.invoke.
+    It may be necessary to read the Payload see the actual response from the Lambda function e.g.,
+
+    ```
+    def succeeded(response):
+        payload = json.loads(response['Payload'].read())
+        # do something with payload
+    ```
+
+    :param function_name: The name of the Lambda function.
+    :type function_name: str
+    :param region_name: AWS region e.g., eu-west-1, ap-southeast-1, etc.
+    :type region_name: str
+    :param payload: The JSON to submit as input to a Lambda function.
+    :type payload: str
+    :param check_success_function: A function to check the Lambda response and determine success or failure.
+    :type check_success_function: function
+    :param log_type: Set to Tail to include the execution log in the response. Otherwise, set to "None".
+    :type log_type: str
+    :param qualifier: A version or alias name for the Lambda.
+    :type qualifier: str
+    :param aws_conn_id: connection id of AWS credentials / region name. If None,
+        credential boto3 strategy will be used
+        (http://boto3.readthedocs.io/en/latest/guide/configuration.html).
+    :type aws_conn_id: str
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        function_name,
+        region_name,
+        payload,
+        check_success_function,
+        log_type="None",
+        qualifier="$LATEST",
+        aws_conn_id=None,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.function_name = function_name
+        self.payload = payload
+        self.check_success_function = check_success_function
+        self.aws_lambda_hook = AwsLambdaHook(
+            function_name, region_name, log_type, qualifier, aws_conn_id=aws_conn_id
+        )
+
+    def execute(self, context):
+        self.log.info("AWS Lambda: invoking {}".format(self.function_name))
+
+        response = self.aws_lambda_hook.invoke_lambda(self.payload)
+
+        try:
+            self._validate_lambda_api_response(response)
```

Comment:
Can you move it to the aws_lambda_hook.invoke_lambda method? I think this function should validate the response.

Reply:
Sure. What are your thoughts on my last comment (about where/how to validate the actual response function; not the response from the Lambda API)

Reply:
Agree with @mik-laj on the first one.

> What are your thoughts on my last comment about where/how to validate the actual response function? Should this operator be doing that (by applying a user supplied function) or is that what a Sensor should be used for (and this operator just executes the Lambda function)

Good question. If you only want to validate a condition / `poke` until it succeeds I would suggest to change it to a Sensor. If you want the actual result of the response -> Operator. 

Also for the operator I would probably not catch the exception at all.
I am also not sure if this is a code smell: catching an exception only to log it and then re-raising it.
```python
        except (AwsLambdaExecutionError, AwsLambdaPayloadError) as e:
            self.log.error(response)
            raise e
```

Reply:
> If you only want to validate a condition / poke until it succeeds I would suggest to change it to a Sensor ... If you want the actual result of the response -> Operator.

This is what I was leaning towards. Would you be happy if I update this operator to remove the `check_success_function`? I don't think it's right.

Presumably any operator downstream from this one would use [XCom](https://airflow.apache.org/concepts.html#xcoms) to receive the response and do something with it?

Reply:
This is going to end up being a pretty trivial operator in the end 😄

Reply:
That's fairly common for operators in Airflow - most of the logic ends up in the hooks.